scripts/metricsgen/metricsgen.go

```diff
@@ -0,0 +1,314 @@
+// metricsgen is a code generation tool for creating constructors for Tendermint
+// metrics types.
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/token"
+	"go/types"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strings"
+	"text/template"
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, `Usage: %[1]s -dir <dir> -struct <struct>
```

Comment:
The `-dir` flag seems not to be used anywhere?

Reply:
Ah, it's used in the tests to direct the tool towards the `testdata` directories.

Reply:
We don't seem to be doing that, or am I missing a file?

For a realistic test, should we just add go:generate tags to the testdata too? That way also we update the test cases when we update the tool.

Reply:
Good idea, updated the tests to check against the contents of the old output.

Reply:
Removed the unused `-dir` flag.